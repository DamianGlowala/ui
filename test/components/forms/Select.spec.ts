import { describe, it, expect } from 'vitest'
import { USelect } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Select', () => {
  it.each([
    ['basic case', {}],
    ['with custom id', { props: { id: 'mySelect' }, options: {} }],
    ['with custom name', { props: { name: 'mySelectInput' }, options: {} }],
    ['with placeholder', { props: { placeholder: 'Select an option' }, options: {} }],
    ['with disabled', { props: { disabled: true }, options: {} }],
    ['with icon', { props: { icon: 'i-heroicons-academic-cap' }, options: {} }],
    ['with loading icon', { props: { loadingIcon: 'i-heroicons-adjustments-horizontal' }, options: {} }],
    ['with leading icon', { props: { leadingIcon: 'i-heroicons-adjustments-vertical' }, options: {} }],
    ['with trailing icon', { props: { trailingIcon: 'i-heroicons-archive-box' }, options: {} }],
    ['with trailing', { props: { trailing: true }, options: {} }],
    ['with leading', { props: { leading: true }, options: {} }],
    ['with loading', { props: { loading: true }, options: {} }],
    ['with padded', { props: { padded: false }, options: {} }],
    ['with custom size', { props: { size: 'lg' }, options: {} }],
    ['with custom color', { props: { color: 'blue' }, options: {} }],
    ['with custom variant', { props: { variant: 'outlined' }, options: {} }],
    ['with custom options', { props: { options: [{ value: 'opt1', label: 'Option 1' }, { value: 'opt2', label: 'Option 2' }] }, options: {} }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Select.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, USelect)
    expect(html).toMatchSnapshot()
  })
})
