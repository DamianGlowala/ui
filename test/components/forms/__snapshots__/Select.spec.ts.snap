// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Select > renders basic case correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class=""></select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with custom color correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class="">
    <option data-v-cc4daab7="" value="opt1">Option 1</option>
    <option data-v-cc4daab7="" value="opt2">Option 2</option>
  </select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 text-red-500 dark:text-red-400 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with custom id correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" id="mySelect" class=""></select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with custom name correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" name="mySelectInput" class=""></select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with custom options correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class="">
    <option data-v-cc4daab7="" value="opt1">Option 1</option>
    <option data-v-cc4daab7="" value="opt2">Option 2</option>
  </select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with custom size correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class=""></select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-3.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with custom variant correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class=""></select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with disabled correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" disabled="" class=""></select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with icon correctly 1`] = `"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class=""></select><span data-v-cc4daab7="" class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-academic-cap flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></span><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span></div>"`;

exports[`Select > renders with leading + icon correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class="">
    <option data-v-cc4daab7="" value="opt1">Option 1</option>
    <option data-v-cc4daab7="" value="opt2">Option 2</option>
  </select><span data-v-cc4daab7="" class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-adjustments-vertical flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></span><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span></div>"
`;

exports[`Select > renders with leading slot + iconName correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class="" iconname="i-heroicons-cloud">
    <option data-v-cc4daab7="" value="opt1">Option 1</option>
    <option data-v-cc4daab7="" value="opt2">Option 2</option>
  </select><span data-v-cc4daab7="" class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5">Leading slot</span><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span></div>"
`;

exports[`Select > renders with leading slot correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class="">
    <option data-v-cc4daab7="" value="opt1">Option 1</option>
    <option data-v-cc4daab7="" value="opt2">Option 2</option>
  </select><span data-v-cc4daab7="" class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5">Leading slot</span><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span></div>"
`;

exports[`Select > renders with leadingIcon correctly 1`] = `"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class=""></select><span data-v-cc4daab7="" class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-adjustments-vertical flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></span><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span></div>"`;

exports[`Select > renders with loading correctly 1`] = `"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class=""></select><span data-v-cc4daab7="" class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-arrow-path-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5 animate-spin"></span></span><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span></div>"`;

exports[`Select > renders with loadingIcon correctly 1`] = `"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class=""></select><span data-v-cc4daab7="" class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-adjustments-horizontal flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5 animate-spin"></span></span><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span></div>"`;

exports[`Select > renders with padded correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class="">
    <option data-v-cc4daab7="" value="opt1">Option 1</option>
    <option data-v-cc4daab7="" value="opt2">Option 2</option>
  </select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with placeholder correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class="">
    <option data-v-cc4daab7="" disabled="" value="">Select an option</option>
  </select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with trailing + icon correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class="">
    <option data-v-cc4daab7="" value="opt1">Option 1</option>
    <option data-v-cc4daab7="" value="opt2">Option 2</option>
  </select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with trailing correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class=""></select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with trailing icon correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class=""></select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span data-v-cc4daab7="" class="i-heroicons-archive-box flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
</div>"
`;

exports[`Select > renders with trailing slot + iconName correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class="" iconname="i-heroicons-bookmark-solid">
    <option data-v-cc4daab7="" value="opt1">Option 1</option>
    <option data-v-cc4daab7="" value="opt2">Option 2</option>
  </select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5">Trailing slot</span>
</div>"
`;

exports[`Select > renders with trailing slot correctly 1`] = `
"<div data-v-cc4daab7="" class=""><select data-v-cc4daab7="" class="">
    <option data-v-cc4daab7="" value="opt1">Option 1</option>
    <option data-v-cc4daab7="" value="opt2">Option 2</option>
  </select>
  <!--v-if--><span data-v-cc4daab7="" class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5">Trailing slot</span>
</div>"
`;
